src/uct/sm/mm/mm_ep.c

```diff
@@ -14,39 +14,79 @@ SGLIB_DEFINE_HASHED_CONTAINER_FUNCTIONS(uct_mm_remote_seg_t,
                                         UCT_MM_BASE_ADDRESS_HASH_SIZE,
                                         uct_mm_remote_seg_hash)
 
+void uct_mm_ep_signal_remote_slow_path_callback(ucs_callbackq_slow_elem_t *self)
+{
+    uct_mm_ep_t *ep = ucs_container_of(self, uct_mm_ep_t, cbq_elem);
+
+    uct_mm_ep_signal_remote(ep, UCT_MM_IFACE_SIGNAL_CONNECT);
+}
 
 /* send a signal to remote interface using Unix-domain socket */
-static ucs_status_t
+ucs_status_t
 uct_mm_ep_signal_remote(uct_mm_ep_t *ep, uct_mm_iface_conn_signal_t sig)
 {
     uct_mm_iface_t *iface = ucs_derived_of(ep->super.super.iface, uct_mm_iface_t);
     int ret;
 
     /**
-     * Send connect message to remote interface
+     * Send connect/disconnect message to remote interface
      */
-    for (;;) {
-        ret = sendto(iface->signal_fd, &sig, sizeof(sig), 0,
-                     (const struct sockaddr*)&ep->cached_signal_sockaddr,
-                     ep->cached_signal_addrlen);
-        if (ret >= 0) {
-            ucs_assert(ret == sizeof(sig));
-            return UCS_OK;
-        } else if (errno == EAGAIN) {
-            /* If sending a signal has failed, retry.
-             * Note the by default the receiver might have a limited backlog,
-             * on Linux systems it is net.unix.max_dgram_qlen (10 by default).
-             */
-            uct_mm_iface_recv_messages(iface);
+    ret = sendto(iface->signal_fd, &sig, sizeof(sig), 0,
+                 (const struct sockaddr*)&ep->cached_signal_sockaddr,
+                 ep->cached_signal_addrlen);
+    if (ret >= 0) {
+        ucs_assert(ret == sizeof(sig));
+        ucs_debug("Sent %sconnect from socket %d to %p",
+                  (sig == UCT_MM_IFACE_SIGNAL_CONNECT) ? "" : "dis", iface->signal_fd,
+                  (const struct sockaddr*)&ep->cached_signal_sockaddr);
+
+        if (ep->cbq_elem_on) {
+            uct_worker_slowpath_progress_unregister(iface->super.worker, &ep->cbq_elem);
+            ep->cbq_elem_on = 0;
+        }
+
+        /* point the ep->fifo_ctl to the remote fifo.
+         * it's an aligned pointer to the beginning of the ctl struct in the remote FIFO */
+        ep->fifo_ctl    = (uct_mm_fifo_ctl_t*) ucs_align_up_pow2
```

Comment:
i think it's better to extract these 3 lines to standalone function, to distinguish the operation 
e.g uct_mm_ep_connected()